### ISSUE TITLE
about date Augmentation，how to dsiable data Augmentation

### ISSUE THREAD

**Issue by yangtao-git** [ultralytics/ultralytics#13379]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

# hsv_h: 0.015 # (float) image HSV-Hue augmentation (fraction)
# hsv_s: 0.7 # (float) image HSV-Saturation augmentation (fraction)
# hsv_v: 0.4 # (float) image HSV-Value augmentation (fraction)
# degrees: 0.0 # (float) image rotation (+/- deg)
# translate: 0.1 # (float) image translation (+/- fraction)
# scale: 0.5 # (float) image scale (+/- gain)
# shear: 0.0 # (float) image shear (+/- deg)
# perspective: 0.0 # (float) image perspective (+/- fraction), range 0-0.001
# flipud: 0.0 # (float) image flip up-down (probability)
# fliplr: 0.5 # (float) image flip left-right (probability)
# bgr: 0.0 # (float) image channel BGR (probability)
# mosaic: 1.0 # (float) image mosaic (probability)
# mixup: 0.0 # (float) image mixup (probability)
# copy_paste: 0.0 # (float) segment copy-paste (probability)
# auto_augment: randaugment # (str) auto augmentation policy for classification (randaugment, autoaugment, augmix)
# erasing: 0.4 # (float) probability of random erasing during classification training (0-0.9), 0 means no erasing, must be less than 1.0.
# crop_fraction: 1.0 # (float) image crop fraction for classification (0.1-1), 1.0 means no c

I don't want to do any scaling, rotation, Mosaic of my data Augmentation operations, how can I adjust the above parameters?

**Comment by glenn-jocher** [comment]
Hello!

To disable the specific data augmentations you mentioned (scaling, rotation, and mosaic), you can adjust the parameters in your configuration file as follows:

- Set `degrees` to `0.0` to disable rotation.
- Set `scale` to `1.0` to keep the image scale unchanged.
- Set `mosaic` to `0.0` to disable mosaic augmentation.

Make sure these values are correctly set in your YAML configuration file before starting your training session. This will ensure that the model trains without applying these augmentations.

If you have any more questions or need further assistance, feel free to ask!

Happy training! 🚀

**Comment by yangtao-git** [comment]
> 你好！
> 
> 要禁用您提到的特定数据增强（缩放，旋转和马赛克），您可以按如下方式调整配置文件中的参数：
> 
> * 设置`degrees`为`0.0`以禁用旋转。
> * 设置`scale`为`1.0`，保持图像比例不变。
> * 设置`mosaic`为`0.0`以禁用马赛克增强。
> 
> 在开始训练之前，请确保在 YAML 配置文件中正确设置这些值。这将确保模型在不应用这些增强的情况下进行训练。
> 
> 如果您还有其他问题或需要进一步的帮助，请随时询问！
> 
> 训练愉快！🚀

I have completed the training and the test results are very good, I have never encountered a detector with 100% test set accuracy like yolov8, thank you for your contribution！
best wishes for you！！

**Comment by glenn-jocher** [comment]
Hello!

We're thrilled to hear about your successful training results and the exceptional accuracy you've achieved with YOLOv8! 🌟 Your feedback is greatly appreciated, and it's fantastic to see such positive outcomes.

If you have any more insights, questions, or if there's anything else we can help you with in your future projects, please don't hesitate to reach out.

Thank you for your kind words and

**Comment by joej970** [comment]
Hi,
can you be more specific about which .yaml  YAML configuration file are you referring to? Also note that I am using Kaggle so I have to update the file from the python notebook each time I run it.

On kaggle, I found one file: '/opt/conda/lib/python3.10/site-packages/ultralytics/cfg/default.yaml'. I update it from my python notebook. Also, when calling train() function I call it with model.train(
    data=YAML_FILE,
    fliplr = 0.0,
    flipud = 0.0,
    translate = 0.0,
    mosaic = 0.0,
  scale = 1.0)

HOWEVER, when I check the train data I still see that the images have been rescaled:

![train_batch24952_51_a00e080e16deb9bfdf54](https://github.com/ultralytics/ultralytics/assets/34656809/fda29888-7c46-4a5e-8228-a2f9addcd73f)

I do not need any augmentation from yolo because I have already augmented the images while generating the dataset.

Thanks!

**Comment by glenn-jocher** [comment]
@joej970 hello,

Thank you for reaching out with your query. The `.yaml` file you're referring to seems to be the default configuration file for Ultralytics YOLO. For custom training configurations, it's recommended to create a separate `.yaml` file specific to your dataset and requirements rather than modifying the default directly.

In your custom `.yaml` file, you can specify all the necessary parameters, including turning off specific augmentations. Here’s a brief example of how your custom `.yaml` might look:

```yaml
# custom_dataset.yaml
train: path/to/train/images
val: path/to/val/images
nc: number_of_classes
names: ['class1', 'class2', ...]

# Augmentation settings
fliplr: 0.0
flipud: 0.0
translate: 0.0
mosaic: 0.0
scale: 1.0
```

Then, load this file in your training script:

```python
model.train(data='custom_dataset.yaml')
```

This should ensure that your specified augmentations, or lack thereof, are correctly applied. If you continue to see unexpected behavior, it might be worth double-checking that the dataset itself hasn't been altered prior to loading and that no other augmentations are specified elsewhere in your code.

Best regards and happy training! 🚀

**Comment by joej970** [comment]
[0;31mRuntimeError[0m: Dataset '/kaggle/input/tarok-ds/yolo_obb_DS_8/data.yaml' error ❌ 'bool' object is not iterable

for data.yaml:

path : ../../input/tarok-ds/yolo_obb_DS_8

train: ../yolo_obb_DS_8/train/images
val: ../yolo_obb_DS_8/valid/images
test: ../yolo_obb_DS_8/test/images

# Train settings -------------------------------------------------------------------------------------------------------
model: # (str, optional) path to model file, i.e. yolov8n.pt, yolov8n.yaml
data: # (str, optional) path to data file, i.e. coco8.yaml
epochs: 100 # (int) number of epochs to train for
time: # (float, optional) number of hours to train for, overrides epochs if supplied
patience: 30 # (int) epochs to wait for no observable improvement for early stopping of training
batch: 16 # (int) number of images per batch (-1 for AutoBatch)
imgsz: 736 # (int | list) input images size as int for train and val modes, or list[w,h] for predict and export modes
save: True # (bool) save train checkpoints and predict results
save_period: -1 # (int) Save checkpoint every x epochs (disabled if < 1)
cache: False # (bool) True/ram, disk or False. Use cache for data loading
device: # (int | str | list, optional) device to run on, i.e. cuda device=0 or device=0,1,2,3 or device=cpu
workers: 8 # (int) number of worker threads for data loading (per RANK if DDP)
project: # (str, optional) project name
name: # (str, optional) experiment name, results saved to 'project/name' directory
exist_ok: False # (bool) whether to overwrite existing experiment
pretrained: True # (bool | str) whether to use a pretrained model (bool) or a model to load weights from (str)
optimizer: auto # (str) optimizer to use, choices=[SGD, Adam, Adamax, AdamW, NAdam, RAdam, RMSProp, auto]
verbose: True # (bool) whether to print verbose output
seed: 0 # (int) random seed for reproducibility
deterministic: True # (bool) whether to enable deterministic mode
single_cls: False # (bool) train multi-class data as single-class
rect: False # (bool) rectangular training if mode='train' or rectangular validation if mode='val'
cos_lr: False # (bool) use cosine learning rate scheduler
close_mosaic: 10 # (int) disable mosaic augmentation for final epochs (0 to disable)
resume: False # (bool) resume training from last checkpoint
amp: True # (bool) Automatic Mixed Precision (AMP) training, choices=[True, False], True runs AMP check
fraction: 1.0 # (float) dataset fraction to train on (default is 1.0, all images in train set)
profile: False # (bool) profile ONNX and TensorRT speeds during training for loggers
freeze: None # (int | list, optional) freeze first n layers, or freeze list of layer indices during training
multi_scale: False # (bool) Whether to use multiscale during training
# Segmentation
overlap_mask: True # (bool) masks should overlap during training (segment train only)
mask_ratio: 4 # (int) mask downsample ratio (segment train only)
# Classification
dropout: 0.0 # (float) use dropout regularization (classify train only)

# Val/Test settings ----------------------------------------------------------------------------------------------------
val: True # (bool) validate/test during training
split: val # (str) dataset split to use for validation, i.e. 'val', 'test' or 'train'
save_json: False # (bool) save results to JSON file
save_hybrid: False # (bool) save hybrid version of labels (labels + additional predictions)
conf: # (float, optional) object confidence threshold for detection (default 0.25 predict, 0.001 val)
iou: 0.7 # (float) intersection over union (IoU) threshold for NMS
max_det: 54 # (int) maximum number of detections per image
half: False # (bool) use half precision (FP16)
dnn: True # (bool) use OpenCV DNN for ONNX inference
plots: True # (bool) save plots and images during train/val

# Augmentation settings
fliplr: 0.0
flipud: 0.0
translate: 0.0
mosaic: 0.0
scale: 1.0
perspective: 0.0 # (float) image perspective (+/- fraction), range 0-0.001
bgr: 0.0 # (float) image channel BGR (probability)
erasing: 0.2 # (float) probability of random erasing during classification training (0-0.9), 0 means no erasing, must be less than 1.0.
box: 7.5 # (float) box loss gain
cls: 2.5 # (float) cls loss gain (scale with pixels)
dfl: 1.5 # (float) dfl loss gain
degrees: 0.0 # (float) image rotation (+/- deg)
hsv_h: 0.015 # (float) image HSV-Hue augmentation (fraction)
hsv_s: 0.4 # (float) image HSV-Saturation augmentation (fraction)
hsv_v: 0.2 # (float) image HSV-Value augmentation (fraction)
bgr: 0.0 # (float) image channel BGR (probability)
names:
 0: 01
 1: 02
 2: 03

**Comment by glenn-jocher** [comment]
@joej970 hello!

Thank you for providing the detailed configuration and error message. It looks like the issue is related to the `data.yaml` file, specifically the line causing the `'bool' object is not iterable` error.

To help us investigate further, could you please provide a minimum reproducible example? This will allow us to better understand the context and reproduce the issue on our end. You can find guidance on creating a minimum reproducible example here: [Minimum Reproducible Example](https://docs.ultralytics.com/help/minimum_reproducible_example).

Additionally, please ensure that you are using the latest versions of `torch` and `ultralytics`. You can upgrade your packages using the following commands:

```bash
pip install --upgrade torch
pip install --upgrade ultralytics
```

From the provided configuration, it seems there might be a syntax issue in your `data.yaml` file. Ensure that all the paths and parameters are correctly formatted. Here’s a simplified example of what your `data.yaml` should look like:

```yaml
path: ../../input/tarok-ds/yolo_obb_DS_8

train: ../yolo_obb_DS_8/train/images
val: ../yolo_obb_DS_8/valid/images
test: ../yolo_obb_DS_8/test/images

names:
  0: 01
  1: 02
  2: 03
```

Make sure there are no extra spaces or incorrect indentations.

If the issue persists, please share the specific lines of code where you are loading and using this `data.yaml` file in your training script. This will help us pinpoint the exact cause of the error.

Looking forward to your response so we can assist you further!

**Comment by Petros626** [comment]
note for me.
